### PR TITLE
Add overlay2 description to overlayfs-driver.md

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Related: https://github.com/docker/docker/pull/22126
CC @dmcgowan 

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
CI failure seems unrelated to my change
```
00:06:45.956 # Filtered (engine) Summary:
00:06:45.956 
00:06:45.956 * link error: (in page engine/getstarted/step_one.md) /docker-for-mac/index.md
00:06:45.956 * link error: (in page engine/getstarted/last_page.md) /docker-for-mac/index.md
00:06:45.956 * link error: (in page engine/installation/mac.md) /docker-for-mac/index.md
00:06:45.956 * link error: (in page engine/installation/windows.md) /docker-for-windows/index.md
00:06:45.956 * link error: (in page engine/getstarted/step_one.md) /docker-for-windows/index.md
00:06:45.956 * link error: (in page engine/getstarted/last_page.md) /docker-for-windows/index.md
00:06:45.956 
00:06:45.956 	Found: 820 files
00:06:45.956 	Found: 6 errors
00:06:45.959 ERRORS = 6
```

File: docs/userguide/storagedriver/overlayfs-driver.md [hunk 1]
```diff
@@ -133,13 +148,85 @@ You can verify all of these constructs from the output of the `mount` command.
...
 The output reflects that the overlay is mounted as read-write ("rw").
 
+
+## Image layering and sharing with OverlayFS (`overlay2`)
+
+While the `overlay` driver only works with a single lower OverlayFS layer and
+hence requires hard links for implementation of multi-layered images, the
+`overlay2` driver natively supports multiple lower OverlayFS layers (up to 128).
+
+The `overlay2` driver consumes less inodes than the `overlay` driver.
```
Comment (cpuguy83) [inline docs/userguide/storagedriver/overlayfs-driver.md]: Maybe also a mention about overlay2 offering significantly better performance for commands that create new layes, such as `docker build`, `docker commit`, and `docker pull`

Comment (dmcgowan) [inline docs/userguide/storagedriver/overlayfs-driver.md]: Related to the speed up for `docker build` and `docker commit`, using the native multiple lower layer support, diffs are "precomputed" by OverlayFS, unlike `overlay` which has to do compare every file against the lower layer to figure out the diff. There should also be a corresponding speed up on `docker push` but I have not done any benchmarks with it.

File: docs/userguide/storagedriver/overlayfs-driver.md [hunk 2]
```diff
@@ -133,13 +148,85 @@ You can verify all of these constructs from the output of the `mount` command.
...
+
+    $ ls /var/lib/docker/overlay2/223c2864175491657d238e2664251df13b63adb8d050924fd1bfcdb278b866f7
+    diff  link  lower  merged  work
+    $ cat /var/lib/docker/overlay2/223c2864175491657d238e2664251df13b63adb8d050924fd1bfcdb278b866f7/lower
+    l/6Y5IM2XC7TSNIJZZFLJCS6I4I4
+    $ ls /var/lib/docker/overlay2/223c2864175491657d238e2664251df13b63adb8d050924fd1bfcdb278b866f7/diff/
+    etc  sbin  usr  var
+
+A directory for running container have similar files and directories as well.
+Note that the lower list is separated by ':'.
```
Comment (dmcgowan) [inline docs/userguide/storagedriver/overlayfs-driver.md]: ordered from lowest layer to upper

**Comment by dmcgowan** [comment]
Thanks @AkihiroSuda for adding this! It is looking really good.

**Comment by AkihiroSuda** [comment]
@cpuguy83 @dmcgowan Thank you for the comments. updated PR.

File: docs/userguide/storagedriver/overlayfs-driver.md (hunk 2 above)
Comment (AkihiroSuda) [inline docs/userguide/storagedriver/overlayfs-driver.md]: you mean, "Note that the lower list is separated by ':', and ordered from _highest_ layer to _lower_", right?

**Comment by vdemeester** [comment]
LGTM 🐯
/cc @thaJeztah

**Comment by SvenDowideit** [comment]
LGTM

**Comment by thaJeztah** [comment]
very nice, thanks!

**Pull request merged**